typescript/23-hello-interview-trees/trees.ts

```diff
@@ -0,0 +1,11 @@
+export class TreeNode {
+  val: number
+  left: TreeNode | null
+  right: TreeNode | null
+
+  constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
```

Comment:
Make arguments required, not optional. This is a bad habit, especially on a core data structures.
Also gets rid of `undefined`